input-gen-runtimes/rt-input-gen.cpp

```diff
@@ -576,6 +622,33 @@ void __inputgen_global(int32_t NumGlobals, VoidPtrTy Global,
   getInputGenRT().registerGlobal(Global, ReplGlobal, GlobalSize);
 }
 
+VoidPtrTy __inputgen_select_fp(VoidPtrTy *PotentialFPs, uint64_t N) {
+  return PotentialFPs[getInputGenRT().registerFunctionPtrIdx(N)];
+}
+
+void __inputgen_access_fp(VoidPtrTy Ptr, int32_t Size, VoidPtrTy Base,
+                          VoidPtrTy *PotentialFPs, uint64_t N) {
+  if (!getInputGenRT().read<FunctionPtrTy *>(Ptr, Base, Size)) {
+    VoidPtrTy FP = PotentialFPs[rand() % N];
```

Comment:
Could you try to put this in the InputGetRT class and use the rand() defined in there, would be nice to have the source of randomness be as contained as possible.